crates/aptos/src/move_tool/disassembler.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use move_command_line_common::files::{
+    MOVE_COMPILED_EXTENSION, MOVE_EXTENSION, SOURCE_MAP_EXTENSION,
+};
+use crate::common::types::{CliCommand, CliTypedResult};
+use std::{fs, path::Path};
+use move_binary_format::binary_views::BinaryIndexedView;
+use move_binary_format::CompiledModule;
+use move_binary_format::file_format::CompiledScript;
+use move_coverage::coverage_map::CoverageMap;
+use move_disassembler::disassembler::{Disassembler, DisassemblerOptions};
+use clap::Parser;
+use async_trait::async_trait;
+use move_bytecode_source_map::{mapping::SourceMapping, utils::source_map_from_file};
+use move_ir_types::location::Spanned;
+
+/// Disassemble the Move bytecode pointed to
+#[derive(Debug, Parser)]
+pub struct AptosDisassembler {
+    /// Skip printing of private functions.
+    #[clap(long = "skip-private")]
+    pub skip_private: bool,
+
+    /// Do not print the disassembled bytecodes of each function.
+    #[clap(long = "skip-code")]
+    pub skip_code: bool,
+
+    /// Do not print locals of each function.
+    #[clap(long = "skip-locals")]
+    pub skip_locals: bool,
+
+    /// Do not print the basic blocks of each function.
+    #[clap(long = "skip-basic-blocks")]
+    pub skip_basic_blocks: bool,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(short = 's', long = "script")]
+    pub is_script: bool,
+
+    /// The path to the bytecode file to disassemble; let's call it file.mv. We assume that two
+    /// other files reside under the same directory: a source map file.mvsm (possibly) and the Move
+    /// source code file.move.
+    #[clap(short = 'b', long = "bytecode")]
```

Comment:
The design as discussed on slack (have you seen this?) with @rahxephon89 was different. Instead of the user providing a path to the file (which lives somewhere at internal places), user just says `aptos move disassemble` and then the entire current package content is diassembled to console.

Reply:
> The design as discussed on slack (have you seen this?) with @rahxephon89 was different. Instead of the user providing a path to the file (which lives somewhere at internal places), user just says `aptos move disassemble` and then the entire current package content is diassembled to console.

But its also OK we stay with this design here... Its a bit more low-level but that functionality is anyway not for mainstream use...

Reply:
That's a good point. This is probably how it should work. This means we'd want to add the standard `--package-dir` flag to point to a different package directory.

If we go this route, we'd probably want to add another flag like `--modules` where you can choose which modules to print, in case you've got a package with heaps of modules in it. So I could use it like `--modules coin account`.

Reply:
Yeah, I think we can keep it as it is and improve it in another PR.